### PR TITLE
Database integrate 

### PR DESCRIPTION
**Pull request by nikitashrestha**
Skill and Contact Section

[database-integrate b47f7d1] database-integrate:#7 - integrate hidden feature with api
[database-integrate cd3e7c6] database-integrate:#7 - use uuid to edit skill
[database-integrate 08cc977] database-integrate:#7 - use uuid and Section
[database-integrate add38f9] database-integrate:#7 - use uuid

### CODE REVIEW

File: src/components/skills/SkillItem.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ const SkillItem = ({ title, values, preview, onHiddenIconClicked, onDelete }) =>
               onClose={toggleEditSkill}
               showModal={editSkill}
               isEdit={editSkill}
-              data={editSkill ? { name: title, subSkills: values } : ''}
+              data={editSkill ? { id: id } : ''}
```
Comment (sagunji) [inline src/components/skills/SkillItem.js]: you can simply have `editSkill ? { id } : ''`

File: src/components/form/skill/AddSkill.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ const AddSkill = ({ onClose, isEdit, values }) => {
       const skills = prevData['skills'];
 
       const index = skills.findIndex(skill => {
-        return skill.name === values.name;
+        return skill.id === values.id;
```
Comment (sagunji) [inline src/components/form/skill/AddSkill.js]:
```suggestion
const index = skills.findIndex(skill => skill.id === values.id)
```
We can remove the `return`

File: src/components/form/skill/AddSkill.js [hunk 3]
```diff
@@ -127,9 +127,15 @@ const AddSkill = ({ onClose, isEdit, values }) => {
 
   const getInitialValues = () => {
     if (isEdit) {
+      const skills = data.get.skills;
+
+      const index = skills.findIndex(skill => {
```
Comment (sagunji) [inline src/components/form/skill/AddSkill.js]:
```suggestion
const index = skills.findIndex(skill => skill.id === values.id)
```
We can remove the `return`

File: src/components/skills/SkillItem.js (hunk 1 above)
Comment (nikitashrestha) [reply src/components/skills/SkillItem.js]: done [database-integrate 43bde83] database-integrate:#7 - pass just id in data param

File: src/components/form/skill/AddSkill.js (hunk 2 above)
Comment (nikitashrestha) [reply src/components/form/skill/AddSkill.js]: done [database-integrate 4ff98e8] database-integrate:#7 - remove return from findIndex

File: src/components/form/skill/AddSkill.js (hunk 3 above)
Comment (nikitashrestha) [reply src/components/form/skill/AddSkill.js]: done [database-integrate 4ff98e8] database-integrate:#7 - remove return from findIndex

File: src/components/form/skill/AddSkill.js [hunk 4]
```diff
@@ -127,9 +125,13 @@ const AddSkill = ({ onClose, isEdit, values }) => {
 
   const getInitialValues = () => {
     if (isEdit) {
+      const skills = data.get.skills;
+
+      const index = skills.findIndex(skill => skill.id === values.id);
```
Comment (sagunji) [inline src/components/form/skill/AddSkill.js]: we can use only find to get us the object

Comment (nikitashrestha) [reply src/components/form/skill/AddSkill.js]: Noted and done : [database-integrate 46f9fd5] use find to locate object

**Review by sagunji — APPROVED** [review]

**Pull request merged**